src/java/main/org/apache/zookeeper/common/X509Util.java

```diff
@@ -214,45 +237,56 @@ public static X509TrustManager createTrustManager(String trustStoreLocation, Str
             tmf.init(new CertPathTrustManagerParameters(pbParams));
 
             for (final TrustManager tm : tmf.getTrustManagers()) {
-                if (tm instanceof X509TrustManager) {
+                if (tm instanceof X509ExtendedTrustManager) {
                     return new X509ExtendedTrustManager() {
-                        HostnameChecker hostnameChecker = HostnameChecker.getInstance(HostnameChecker.TYPE_TLS);
+                        X509ExtendedTrustManager x509ExtendedTrustManager = (X509ExtendedTrustManager) tm;
```

Comment:
Do you think extending X509ExtendedTrustManager outside will help us test it better?.